### PR TITLE
framework/task_manager: Add resource collection logic

### PR DESCRIPTION
**Pull request by jeongchanKim**
When request to terminate or restart, resource collection is needed.
So if user registered the resource collection handler, task_manager will call that handler.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [255b2902eb2b003e89fc873694ecf37136039e2b] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [255b2902eb2b003e89fc873694ecf37136039e2b] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [3bef4d087ab6a26263c6cfbcae1d6fbd4a99fd86] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [3bef4d087ab6a26263c6cfbcae1d6fbd4a99fd86] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [97c7771a3193cb5975aa71201ccd4196609eb323] - Code Rule Check Result:
framework/src/task_manager/taskmgr_interface.c:458: ERROR: [SPC_M_KWD] space required before the open parenthesis '('

**Comment by seinfra** [comment]
Target : [97c7771a3193cb5975aa71201ccd4196609eb323] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [def7ccab34568d76dc1c162492fbd1e6c24c7406] -

**Comment by seinfra** [comment]
Target : [def7ccab34568d76dc1c162492fbd1e6c24c7406] -

**Comment by seinfra** [comment]
Target : [0d9a5d039ed4ff021558ad377e62404de19017ee] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [0d9a5d039ed4ff021558ad377e62404de19017ee] - Code Rule Check Result:
framework/src/task_manager/taskmgr_interface.c:456: ERROR: [SPC_M_KWD] space required before the open parenthesis '('

**Comment by seinfra** [comment]
Target : [7be3749401843f67283ed4cfe7b8cd2dbba25a96] -

**Comment by seinfra** [comment]
Target : [7be3749401843f67283ed4cfe7b8cd2dbba25a96] -

File: framework/src/task_manager/taskmgr_interface.c [hunk 1]
```diff
@@ -451,6 +451,18 @@ int task_manager_set_handler(void (*func)(int signo, tm_msg_t *data))
 	return OK;
 }
 
+int task_manager_set_res_handler(void (*func)(int signo, tm_msg_t *data))
+{
+	if (func == NULL) {
+		return TM_INVALID_PARAM;
+	}
+
+	if(atexit((void *)func) != OK) {
```
Comment (sunghan-chang) [inline framework/src/task_manager/taskmgr_interface.c]:
```
if (
```

File: framework/src/task_manager/taskmgr_interface.c [hunk 2]
```diff
@@ -451,6 +451,18 @@ int task_manager_set_handler(void (*func)(int signo, tm_msg_t *data))
 	return OK;
 }
 
+int task_manager_set_res_handler(void (*func)(int signo, tm_msg_t *data))
```
Comment (sunghan-chang) [inline framework/src/task_manager/taskmgr_interface.c]: It is difficult to distinguish set_handler and set_res_handler.
Please rename set_handler to know the purpose of function.

**Comment by seinfra** [comment]
Target : [afdb236d159965a07f6ebc40bf07d7ec75ab4d05] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [afdb236d159965a07f6ebc40bf07d7ec75ab4d05] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [726d057029edd622f5550c0b0aaec3971bef40f2] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [726d057029edd622f5550c0b0aaec3971bef40f2] - Code Rule Check (C++) OK.

**Pull request merged**